tools/cache_creator/CMakeLists.txt

```diff
@@ -0,0 +1,53 @@
+##
+ #######################################################################################################################
+ #
+ #  Copyright (c) 2020 Google LLC. All Rights Reserved.
+ #
+ #  Permission is hereby granted, free of charge, to any person obtaining a copy
+ #  of this software and associated documentation files (the "Software"), to deal
+ #  in the Software without restriction, including without limitation the rights
+ #  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ #  copies of the Software, and to permit persons to whom the Software is
+ #  furnished to do so, subject to the following conditions:
+ #
+ #  The above copyright notice and this permission notice shall be included in all
+ #  copies or substantial portions of the Software.
+ #
+ #  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ #  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ #  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ #  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ #  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ #  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ #  SOFTWARE.
+ #
+ #######################################################################################################################
+add_library(cache_creator_lib INTERFACE)
+
+list(APPEND CMAKE_MODULE_PATH "${XGL_LLVM_BUILD_PATH}/lib/cmake/llvm")
+include(LLVMConfig)
+
+llvm_map_components_to_libnames(llvm_libs Support)
+target_link_libraries(cache_creator_lib INTERFACE pal ${llvm_libs})
+
+target_include_directories(cache_creator_lib INTERFACE
+    ${XGL_LLVM_SRC_PATH}/include
+    ${LLVM_INCLUDE_DIRS}
+)
+
+get_target_property(XGL_COMPILE_OPTIONS xgl COMPILE_OPTIONS)
+get_target_property(XGL_COMPILE_DEFINITIONS xgl COMPILE_DEFINITIONS)
+
+# Compile with the same flags and definitions as the ICD and LLVM.
+target_compile_definitions(cache_creator_lib INTERFACE ${LLVM_DEFINITIONS})
+target_compile_definitions(cache_creator_lib INTERFACE ${XGL_COMPILE_DEFINITIONS})
+target_compile_options(cache_creator_lib INTERFACE ${XGL_COMPILE_OPTIONS})
+
+target_sources(cache_creator_lib INTERFACE cache_creator.cpp)
+
+add_executable(cache-creator)
```

Comment:
Add a brief comment explaining what this executable is intended to be.

It's good practice and can really assist build engineers.